lib/structures/CommandInteraction.js

```diff
@@ -0,0 +1,234 @@
+*/
+class CommandInteraction extends Interaction {
+    constructor(data, client) {
+        super(data, client);
+
+        if(data.channel_id !== undefined) {
+            this.channelID = data.channel_id;
+        }
+
+        this.data = data.data;
+
+        if(data.guild_id !== undefined) {
+            this.guildID = data.guild_id;
+        }
+
+        if(data.member !== undefined) {
+            this.member = new Member(data.member, client.guilds.get(data.guild_id), this._client);
+        }
+
+        if(data.user !== undefined) {
+            this.user = new User(data.user, this._client);
+        }
+
+    }
+
+    /**
+    * Acknowledges the interaction with a defer response
+    * Note: You can **not** use more than 1 initial interaction response per interaction.
+    * @arg {Number} [flags] 64 for Ephemeral
+    * @returns {Promise}
+    */
+    async acknowledge(flags) {
+        return this._client.createInteractionResponse.call(this._client, this.id, this.token, {
+            type: InteractionResponseTypes.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+            data: {
+                flags: flags || 0
+            }
+        });
+    }
+
+    /**
+    * Respond to the interaction with a followup message
+    * @arg {String | Object} content A string or object. If an object is passed:
+    * @arg {Object} [content.allowedMentions] A list of mentions to allow (overrides default)
+    * @arg {Boolean} [content.allowedMentions.everyone] Whether or not to allow @everyone/@here.
+    * @arg {Boolean | Array<String>} [content.allowedMentions.roles] Whether or not to allow all role mentions, or an array of specific role mentions to allow.
+    * @arg {Boolean | Array<String>} [content.allowedMentions.users] Whether or not to allow all user mentions, or an array of specific user mentions to allow.
+    * @arg {String} content.content A content string
+    * @arg {Object} [content.embeds] An array of up to 10 embed objects. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/channel#embed-object) for object structure
+    * @arg {Number} [content.flags] 64 for Ephemeral
+    * @arg {Object | Array<Object>} [content.file] A file object (or an Array of them)
+    * @arg {Buffer} content.file.file A buffer containing file data
+    * @arg {String} content.file.name What to name the file
+    * @arg {Boolean} [content.tts] Set the message TTS flag
+    * @returns {Promise<Message?>}
+    */
+    async createFollowup(content) {
+        if(content !== undefined) {
+            if(typeof content !== "object" || content === null) {
+                content = {
+                    content: "" + content
+                };
+            } else if(content.content !== undefined && typeof content.content !== "string") {
+                content.content = "" + content.content;
+            }
+        }
+        return this._client.executeWebhook.call(this._client, this.applicationID, this.token, content);
```

Comment:
This should either be converted into its own client function, or wait should be added so executeWebhook actually constructs the message (it doesn't attempt it unless await is true)

(I think we use Object.assign instead of spread around here)
```suggestion
        return this._client.executeWebhook.call(this._client, this.applicationID, this.token, Object.assign({ wait: true }, content);
```